core/src/main/java/org/wildfly/channel/ChannelImpl.java

```diff
@@ -222,16 +224,16 @@ Optional<ResolveLatestVersionResult> resolveLatestVersion(String groupId, String
                     Set<String> versions = resolver.getAllVersions(groupId, artifactId, extension, classifier);
                     final Optional<String> latestVersion = versions.stream().sorted(COMPARATOR.reversed()).findFirst();
                     if (latestVersion.isPresent()) {
-                        return Optional.of(new ResolveLatestVersionResult(latestVersion.get(), this));
+                        return Optional.of(new ResolveLatestVersionResult(latestVersion.get(), this, null));
```

Comment:
should we warn that artifacts resolved via the fallback mechanism are not integrity-checked?

Reply:
I am not sure we should warn. I updated the spec to make it clear that no check is done in the fall-back cases.